packages/main/src/RPA/Dialogs.py

```diff
@@ -0,0 +1,428 @@
+            checkedvalue = ""
+            if "default" in item and item["default"] == option:
+                checkedvalue = " checked"
+            formhtml += f"""<input type=\"radio\" id=\"{option}\" name=\"{item['id']}\"
+                            value="{option}"{checkedvalue}>
+                            <label for=\"{option}\">{option}</label><br>"""
+        return formhtml
+
+    def get_checkbox(self, item):
+        formhtml = ""
+        if "label" in item:
+            formhtml += f"<p>{item['label']}</p>"
+        idx = 1
+        for option in item["options"]:
+            formhtml += f"""<input type=\"checkbox\" id=\"{item['id']}{idx}\"
+                                name=\"{item['id']}{idx}\" value="{option}">
+                                <label for=\"{item['id']}{idx}\">{option}</label><br>"""
+            idx += 1
+        return formhtml
+
+    def get_dropdown(self, item):
+        formhtml = (
+            f"<label for=\"{item['id']}\">{item['label']}</label><br>"
+            f"<select name=\"{item['id']}\" name=\"{item['id']}\"><br>"
+        )
+        for option in item["options"]:
+            selected = ""
+            if "default" in item and item["default"] == option:
+                selected = " selected"
+            formhtml += f'<option name="{option}"{selected}>{option}</option>'
+        formhtml += "</select><br>"
+        return formhtml
+
+    def get_submit(self, item):
+        formhtml = ""
+        for button in item["buttons"]:
+            formhtml += f"""<input type=\"submit\" name=\"{item['name']}\"
+                        value=\"{button}\">"""
+        return formhtml + "<br>"
+
+    def get_textarea(self, item):
+        defaulttext = item["default"] if "default" in item else ""
+        return f"""<textarea name=\"{item['name']}\" rows=\"{item['rows']}\"
+                    cols=\"{item['cols']}\">{defaulttext}</textarea><br>"""
+
+    def get_textinput(self, item):
+        return (
+            f"<label for=\"{item['name']}\">{item['label']}</label><br>"
+            f"<input type=\"text\" name=\"{item['name']}\"><br>"
+        )
+
+    def get_fileinput(self, item):
+        accept_filetypes = (
+            "accept=\"{item['filetypes']}\"" if "filetypes" in item.keys() else ""
+        )
+        formhtml = (
+            f"<label for=\"{item['name']}\">{item['label']}</label><br>"
+            f"<input type=\"file\" id=\"{item['id']}\" "
+            f"name=\"{item['name']}\" {accept_filetypes} multiple><br>"
+        )
+        if "target_directory" in item.keys():
+            formhtml += (
+                f"<input type='hidden' name='target_directory'"
+                f" value=\"{item['target_directory']}\">"
+            )
+        return formhtml
+
+    def get_title(self, item):
+        return f"<h3>{item['value']}</h3>"
+
+    def get_text(self, item):
+        return f"<p>{item['value']}</p>"
+
+    def create_form(self, message):
```

Comment:
This HTML generator should probably be separate, and I think we have bs4 already as a dependency which would have made this a bit easier. But no need actions needed for now, just keep in mind if we keep working on this (a lot) more.

Reply:
I was thinking to make it separate but did not want to go there at this stage. Good point on bs4 - did not notice that it is part of our dependencies.